src/modules/toolbox/components/measureToolContent/ShareMeasureDialog.js

```diff
@@ -0,0 +1,93 @@
+import { html } from 'lit-html';
+import { BaElement } from '../../../BaElement';
+import { $injector } from '../../../../injection';
+import clipboardIcon from './assets/clipboard.svg';
+import shareIcon from './assets/share.svg';
+import css from './shareMeasureDialog.css';
+
+/**
+ * A content component to show and share perma-links of 
+ * user-generated measurement-data
+ * @class
+ * @author thiloSchlemmer
+ * @author alsturm
+ */
+export class ShareMeasureDialog extends BaElement {
+
+	constructor() {
+		super();
+		const { TranslationService: translationService, EnvironmentService: environmentService, ShareService: shareService } = $injector.inject('TranslationService', 'EnvironmentService', 'ShareService');
+		this._translationService = translationService;
+		this._environmentService = environmentService;
+		this._shareService = shareService;
+		this._shareUrls = null;
+	}
+
+
+	createView() {
+		const translate = (key) => this._translationService.translate(key);
+
+		if (this._shareUrls) {
+			const useShareApi = this._environmentService.getWindow().navigator.share ? true : false;
+
+			const editableContent = this._buildShareItem(this._shareUrls.adminId, translate('toolbox_measureTool_share_link_readonly'), useShareApi);
+			const readOnlyContent = this._buildShareItem(this._shareUrls.fileId, translate('toolbox_measureTool_share_link_edit'), useShareApi);
+
+			return html`
+			<style>${css}</style>
+            <div class='share_content'>
+                ${editableContent}
+                ${readOnlyContent}
+            </div>`;
+		}
+
+		return html.nothing;
+	}
+
+	_buildShareItem(url, label, useShareApi) {
+		const translate = (key) => this._translationService.translate(key);
+		const onCopyUrlToClipBoard = async () => this._copyValueToClipboard(url);
+
+		const getShareApiContent = (useShareApi) => {
+			if (useShareApi) {
+				const onClickWithApi = () => {
+					this._environmentService.getWindow().navigator.share({
+						title: translate('toolbox_measureTool_share_link_title'),
+						url: url
+					}).catch(console.error);
```

Comment:
`catch` does take a function as an argument.
 It also seems that the test is missing for that case